app/repositories/sensor/sensor.repository.go

```diff
@@ -0,0 +1,141 @@
+package sensor
+
+import (
+	"context"
+
+	model "database-ms/app/models"
+	"database-ms/config"
+
+	mgo "gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+)
+
+// SensorRepository, used to perform DB operations
+// Interface contains basic operations on sensor document
+// So that, DB operation can be performed easily
+type SensorRepository interface {
+
+	// Create, will perform db operation to save sensor
+	// Returns new id and error if occurs
+	Create(context.Context, *model.Sensor) error
+
+	// FindByThingId, find all sensor belonging to a thing
+	// It will retrun error if occurs
+	FindByThingId(context.Context, string) ([]*model.Sensor, error)
+
+	// FindOneById, find the sensor by the provided id
+	// return matched sensor and error if any
+	FindOneById(context.Context, string) (*model.Sensor, error)
+
+	// FindOneByThingIdAndSid
+	FindOneByThingIdAndSid(context.Context, string, int) (*model.Sensor, error)
+
+	// FindByThingIdAndLastUpdate
+	FindByThingIdAndLastUpdate(context.Context, string, int64) ([]*model.Sensor, error)
+
+	// FindOne
+	FindOne(context.Context, interface{}) (*model.Sensor, error)
```

Comment:
Why is this needed? We will only be fetching all sensors for a thing ID, or a sensor by it's ID.